.github/workflows/check-catalyst.yaml

```diff
@@ -343,24 +320,31 @@ jobs:
       - name: Build Runtime test suite for Lightning simulator
         if: ${{ matrix.simulator == 'lightning' }}
         run: |
-            cmake -S runtime -B build -G Ninja \
-                  -DCMAKE_BUILD_TYPE=Debug \
-                  -DENABLE_LIGHTNING_KOKKOS=OFF \
-                  -DLIGHTNING_GIT_TAG=$LIGHTNING_GIT_TAG_VALUE \
-                  -DQIR_STDLIB_LIB="$(pwd)/runtime/qir-stdlib-build/" \
-                  -DQIR_STDLIB_INCLUDES="$(pwd)/runtime/qir-stdlib-build/include/" \
-                  -DENABLE_CODE_COVERAGE=ON
-            cmake --build build --target runner_tests -j$(nproc)
+            pushd runtime
+            C_COMPILER=$(which gcc) \
+            CXX_COMPILER=$(which g++) \
+            RT_BUILD_DIR=../build \
+            LIGHTNING_GIT_TAG_VALUE=$LIGHTNING_GIT_TAG_VALUE \
+            QIR_STDLIB_DIR="$(pwd)/qir-stdlib-build/" \
+            QIR_STDLIB_INCLUDES_DIR="$(pwd)/qir-stdlib-build/include/" \
+            CODE_COVERAGE=ON \
+            make test
```

Comment:
Yes great point, since we are using the makefile there is no need to invoke these additional commands separately 💯 

The only thing is `make coverage` should take care of the `CODE_COVERAGE=ON` flag no? Since we have the two targets `make test` and `make coverage`, there probably isn't a need to expose this flag externally.

Reply:
I prefer to have `CODE_COVERAGE` exposed externally, but we can define it to be `OFF` by default and `ON` by default for the `coverage` target. In fact, that is already the case:

```
coverage: CODE_COVERAGE=ON
```

Reply:
Got it thanks 👍 Then we can apply your fix above and invoke `make coverage` :)

Reply:
Took a bit longer to implement than expected due to dealing with caching of the qir std-lib and CMakeCache.